pyk/src/pyk/kore/rpc.py

```diff
@@ -291,23 +291,24 @@ def close(self) -> None:
         self._transport.close()
 
     def request(self, method: str, **params: Any) -> dict[str, Any]:
-        old_id = self._req_id
+        # Generate unique ID regardless of different clients, repeated requests from the same client, etc.
+        uid = hash_str((id(self), self._req_id))
```

Comment:
Does this mean that the requests will no longer be linear? If they are not, is there any way of establishing the correct order in the bug reports?

Reply:
My understanding is that the order is determined by the ordering of the commands files, which are bash scripts that refer to the request files, and those are still named using a counter. This would only be changing the names of the request files (and the IDs of the requests referred to in the requests themselves)
See: https://github.com/runtimeverification/k/blob/0cfdcda602d13ef43f8fbc508c0e405dd0a5784e/pyk/src/pyk/utils.py#L541